aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -71,10 +71,48 @@ void erfinv_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+void digamma_kernel_cuda(TensorIterator& iter) {
+  if (iter.dtype() == kHalf) {
+    gpu_kernel(iter, []GPU_LAMBDA(at::Half a) -> at::Half {
+        return calc_digamma<float>(a);
+    });
+  } else {
+    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "digamma_cuda", [&]() {
+      gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
+        return calc_digamma(a);
+      });
+    });
+  }
+}
+
+void trigamma_kernel_cuda(TensorIterator& iter) {
+  if (iter.dtype() == kHalf) {
+    gpu_kernel(iter, []GPU_LAMBDA(at::Half a) -> at::Half {
+        return calc_trigamma<float>(a);
```

Comment:
We don't need this special casing anymore because the function handles the conversion to accscalar_t and back.

Reply:
Resolved